app/js/lib/form_validation.js

```diff
@@ -83,7 +87,12 @@ var jQuery = require( 'jquery' ),
 		validationUrl: '',
 		sendFunction: null,
 		validate: function ( formValues ) {
-			var postData = {
+			var amountAsFloat = parseFloat( formValues.amount ),
+				postData;
+			if ( amountAsFloat === 0 || isNaN( amountAsFloat ) || !formValues.paymentType ) {
```

Comment:
Since we use the German locale for the amount, this only works, because the minimum donation value is 1 Euro.